app/views/venues/show.html.erb

```diff
@@ -1,15 +1,27 @@
 <h1><%= @venue.venue_name %></h1>
 <p><%= @venue.description %></p>
+
+<% unless current_user.nil? %>
+  <% if current_user.admin? || @venue.user_id == current_user.id %>
+    <%= button_to "Delete Venue", venue_path, class: "button", method: "delete", label: "Delete Venue"%>
+  <% end %>
+<% end %>
 <ul>
```

Comment:
There's starting to be a lot of logic in this view - consider abstracting out the logic of whether a user can delete a resource:

**show.html.erb**
```
<%= if @venue.deletable_by(current_user) %>
  <%= button_to ... %>
<% end %>
```

**venue.rb**
```
def deletable_by(user)
  user ? (user.admin || self.user == user) : false
end
```